README.md
# wabii
Maquetado ecommerce


CHANGELOG.md
# Change Log
El formato de este documento está basado en [Keep a Changelog]
(http://keepachangelog.com/) y usa versionado semántico [http://semver.org/].

## [2.3.6] - 2018-06-27
### Added
- Se agrego GIT.md con la guia de utilización de GIT y GITLAB

## [2.3.5] - 2018-05-28
### Added
- Se agrego el SMTP para el envio de formularios

## [2.3.4] - 2017-29-11
### Added
- Enlace para compartir en LinkedIn con el helper compartirUrl()

## [2.3.3] - 2017-08-22
### Changed
- Se modificó la funcionalidad para traducir los sitios con el helper

## [2.3.2] - 2017-08-15
### Added
- Se agregó el gulp-autoprefixer a gulpfile.js


## [2.3.1] - 2017-08-11
### Fixed
- Se ajustó el .htaccess /sitio-en-mantenimiento para que redirija bien. 
  No usar /sitio-en-mantenimiento en el dir raíz.
- Cambié la forma en que se minificaba el JS para que no genere problemas cuando 
  el dueño del JS y el dueño del script no coinciden


## [2.3] - 2017-07-27
### Added
- Agrego las propiedades "titulo" y "thumbnail" al modelo Archivo
- Nuevo helper enlaceTel() que transforma un teléfono en un enlace

### Fixed
- Actualizo el link a jQuery para que funcione servido con HTTPS
- Ajusto el minificador para que no intente comprimir ni modificar los tiempos de caché, 
  eso lo manejo todo desde el .htaccess
- El campo fecha ya no tiene un valor siempre. Antes para hacer búsquedas por otros campos 
  había que borrar primero el campo tipo fecha.
  (core\views\includes\lista\buscador.php)
- Filtro los listados por categoría (categ)
  (core\views\includes\lista\buscador.php)

### Changed
- Se ajustó sftp-config.json para que no suba la carpetas node_modules
- Se ajustó package.json para que tenga nombre genérico el proyecto y la descripción
- El tiempo por defecto de caché ahora es un año, como sugiere la herramienta de 
  auditoría del Chrome Devtools.
- El CSS de los plugins JS se agrega en la cabecera en vez del body para poder 
  sobreescribirlos desde el CSS principal.
- Habilito algunas funciones de seguridad en el .htaccess y fuerzo las 
  redireccines a HTTPS
- Aumento a un año el tiempo de expiración en caché de las imágenes y las fuentes
- Los helpers ahora se cargan automáticamente
- Quito la opción "default" del modelo Foto y ya no devuelvo nada si no existe la foto
- Agrego el parametro $tipo al método Video::thumbnail()


## [2.2.3] - 2017-07-04
### Fixed
- Ahora sí se toma el valor por defecto de las metas
- Ahora se agrega bien el replyTo de los formularios

### Changed
- Ajusté el gulpfile.js para que tire los errores solo una vez y ajusté el mensaje de error.
	También evito que la tarea SASS tire error cuando error de variable sin definir.
- Ya no hay que poner los datos de conexión FTP en el gulpfile.js, ahora se leen desde sftp-config.json
- Actualizé el PHPMailer a la versión 5.2.23


## [2.2.2] - 2017-07-02
### Fixed
- Ajusto el .htaccess para que se resuelvan como .php los archivos sin extensión de un subdirectorio


## [2.2.1] - 2017-06-30
### Fixed
- Cierro el <main> en footer.php


## [2.2] - 2017-06-16
### Added
- Nuevo recurso para cargar las fuentes tipográficas y mejorar la velocidad de carga del sitio. 


## [2.1.2] - 2017-05-12 -cambiado por Fran que no entiende la lógica. JOJO
### Fixed
- api/forms/core/formulario.php -> se cambió la ruta por esta (API_PATH.'forms/libs/PHPMailer/PHPMailerAutoload.php'); (linea 214) 


## [2.1.1] - 2017-05-12
### Changed
- api/forms/core/campos/Captcha.php -> se cambió la ruta de los 'require_once' en el case de 'requerido' (linea 36 - linea 41) 


## [2.1] - 2017-05-03
### Added
- Ahora se puede definir qué estrategias se usan para detectar el idioma y 
en qué orden (detección por IP, por URL, desde una cookie o por el navegador).
- Agrego documentación sobre el uso del helper "Traduccion".
- Agrego los parámetos "theme" y "size" al método html() del campo
 Captcha.
- Compruebo la versión mínima de PHP del servidor

### Fixed
- Las metas de la configuración no se estaban pudiendo sobreescribir
- El helper "Traduccion" no estaba funcionando. Cambio el alias _() por __().
- Ajusto las rutas de todos los archivos del formulario
- El formulario ahora la la configuración desde api/config/forms.php

### Changed
- Mejoro la documentación del helper Url con ejemplos.


## [2.0] - 2017-04-20
### Added
- Soporte para SASS con Gulp

### Changed
- Separación del maquetado de la programación. Ahora la programación está en /api/.
- Centralización de las configuraciones. Todo se configura desde /api/config/.
- Centralización de las metas. Ahora se configuran desde /api/config/metas.php.
- Actualización de README con instrucciones de uso.
- Actualización del .htaccess a la última versión de H5TB


## [1.1] - 2017-02-13
### Security
- Bloqueo el acceso al CHANGELOG.

### Added
- Nuevo README con instrucciones de uso. En el .htaccess bloqueo el acceso a 
este archivo.
- Agrego sitio-en-mantenimiento.php para bloquear el acceso al sitio durante la 
publicación.
- Nuevo helper enlaceCompartir() para generar enlaces para compartir en redes sociales.

### Changed
- Actualizo el .htaccess con la última versión de las configuraciones de 
servidor de HTML5 Boilerplate [https://github.com/h5bp/server-configs-apache]
- Muevo el CSS de boostrap a /css/bootstrap/

### Fixed
- Dejo de usar el servicio Clousure de Google para minificar los recursos
(empezó a funcionar mal)


## [1.0.0] - 2016-11-11
### Added
- Agrego CHANGELOG.md
- Nueva forma de organizar el JS. Con el nuevo helper PHP "Plugins" se genera un
array con nombres de funcionalidades JS  a cargar. Todo el JS ahora está en 
main.js y se va cargando junto con su CSS según se active la funcionalidad o no.
- El captcha ya no depende de jQuery para generarse

### Changed
- Remuevo del CSS y JS de Boostrap todo lo que no se usa.
  [http://getbootstrap.com/customize/?id=7298e6f1ebdbdb6a3cd32d1b43dd5118]
- Actualizo jQuery a la versión 2.2.4
- Actualizo WOW a la versión 2.2.4, agrego valores por defecto y desactivo la 

gulpfile.js
const { parallel, src, dest, watch } = require("gulp");
const fs = require("fs");

const sass = require("gulp-sass");
sass.compiler = require("node-sass");

const autoprefixer = require("gulp-autoprefixer");
const gutil = require("gulp-util");
const notify = require("gulp-notify");
const plumber = require("gulp-plumber");
const RJSON = require("relaxed-json");
const ftp = require("vinyl-ftp");

/**
 * SASS
 * Compila los archivo SASS en un archivo CSS
 */
function sassCompile() {
  return src("./sass/main.sass")
    .pipe(
      plumber({
        errorHandler: reportError,
      })
    )
    .pipe(sass({ outputStyle: "expanded" }))
    .pipe(
      autoprefixer({
        cascade: false,
      })
    )
    .pipe(dest("sass/"));
}

function sassWatchAndCompile() {
  watch("./sass/**/*.sass", { delay: 500 }, sassCompile);
}

/**
 * Sube al servidor los archivos actualizados
 * Mira los cambios en los archivos locales (solo SASS) y los copia al servidor cuando se detecta un cambio
 */
function sassWatchAndDeploy() {
  var ftpConfig = loadFtpConfig();
  // var conn = ftp.create({
  //   host: ftpConfig.host,
  //   user: ftpConfig.user,
  //   password: ftpConfig.password,
  //   parallel: 3,
  //   log: gutil.log,
  // });

  const watcher = watch(["sass/**/*", "sass/*"]);
  watcher.on("change", function (path) {
    console.log(
      'Bien joven Padawan, has realizado cambios! Vamos a subir los files de la estrella de la muerte "' +
        path
    );

    // return src([path], { base: ".", buffer: false }).pipe(
    //   conn.dest(ftpConfig.remote_path)
    // );
  });
}

/**
 * Carga configuración FTP
 * Cargo la configuración del archivo del plugin SFTP del Sublime
 * https://github.com/davidtheclark/cosmiconfig
 */
function loadFtpConfig() {
  return RJSON.parse(
    RJSON.transform(fs.readFileSync("./sftp-config.json", "utf8"))
  );
}

/**
 * Manejador de errores
 * Muestra los mensajes de error como una notificación
 * https://github.com/mikaelbr/gulp-notify/issues/81#issuecomment-100422179
 */
function reportError(error) {
  var lineNumber = error.lineNumber ? "LINE " + error.lineNumber + " -- " : "";

  notify({
    title: "Error [" + error.plugin + "]",
    message: lineNumber + "<%= error.message %>",
    sound: "Sosumi", // See: https://github.com/mikaelbr/node-notifier#all-notification-options-with-their-defaults
  }).write(error);

  gutil.beep(); // Beep 'sosumi' again

  // Prevent the 'watch' task from stopping
  this.emit("end");
}

exports.sassCompile = sassWatchAndCompile;
exports.default = parallel(sassWatchAndCompile, sassWatchAndDeploy);


includes/head.php
<?php

$metas = Metas::obtener(SECCION);
$empresa = Config::obtener('empresa');

?>
<!DOCTYPE html>
<html lang="<?= IDIOMA ?>">
  <head>
    <meta charset="utf-8">

    <title><?= $metas->titulo?></title>
    <meta name="description"            content="<?= $metas->descripcion;?>"/>

    <!-- Twitter Card data -->
    <meta name="twitter:card"           content="summary_large_image"/>
    <?php if(isset($empresa->redes['tw']) AND $empresa->redes['tw']){ ?>
    <meta name="twitter:site"           content="@<?= $empresa->redes['tw'] ?>"/>
    <?php } ?>
    <meta name="twitter:title"          content="<?= $metas->titulo ?>" />
    <meta name="twitter:description"    content="<?= $metas->descripcion;?>" />
    <meta name="twitter:image"          content="<?= $metas->img ?>" />

    <!-- Open Graph data -->
    <meta property="og:title"           content="<?= $metas->titulo ?>" />
    <meta property="og:description"     content="<?= $metas->descripcion;?>"/>
    <meta property="og:image"           content="<?= $metas->img; ?>" />
    <meta property="og:site_name"       content="<?= $empresa->nombre ?>"/>
    <meta property="og:url"             content="<?= Url::actual(); ?>"/>

    <meta name="robots"   content="all">

    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />

    <base href="<?= BASE_URL ?>">

    <link rel="canonical" href="<?= Url::actual() ?>" />
    <?php if( Idioma::deteccionHabilitada() ){
      foreach(Idioma::idiomasDisponibles() AS $idioma){
          echo '<link rel="alternate" hreflang="'.$idioma.'" href="'.Url::actual($idioma).'" />'."\r\n";
      }
    } ?>
    <link href="https://fonts.googleapis.com/icon?family=Material+Icons"
      rel="stylesheet">
      <link href="https://fonts.googleapis.com/icon?family=Material+Icons+Outlined"
      rel="stylesheet">
    <!-- Favicons: http://realfavicongenerator.net/ -->
    <link rel="apple-touch-icon" sizes="180x180" href="<?=BASE_URL?>images/favicons/apple-touch-icon.png">
    <link rel="icon" type="image/png" href="<?=BASE_URL?>images/favicons/favicon-32x32.png" sizes="32x32">
    <link rel="icon" type="image/png" href="<?=BASE_URL?>images/favicons/favicon-16x16.png" sizes="16x16">
    <link rel="manifest" href="<?=BASE_URL?>images/favicons/manifest.json">
    <link rel="mask-icon" href="<?=BASE_URL?>images/favicons/safari-pinned-tab.svg" color="#da802d">
    <link rel="shortcut icon" href="<?=BASE_URL?>images/favicons/favicon.ico">
    <meta name="msapplication-config" content="<?=BASE_URL?>images/favicons/browserconfig.xml">
    <meta name="theme-color" content="#ffffff">
    <link href="https://fonts.googleapis.com/css2?family=Rubik:wght@300;400;500;700&display=swap" rel="stylesheet">
    <?php
   
    // Uno y minifico todos los CSS comunes a todas las páginas
    $main_css = $minified->merge(BASE_PATH.'css/main.min.css', 'css', array(
          BASE_PATH.'css/bootstrap/bootstrap.min.css',
          BASE_PATH.'sass/main.css',
      ));
    echo ' <link rel="stylesheet" href="'.str_replace(BASE_PATH, '', $main_css).'?v='.filemtime($main_css).'" />'."\r\n";
    ?>
    
    <!-- Carga las tipografías -->
    <script async src="js/fonts.js"></script>

    <!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
      <script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
      <script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
    <![endif]-->
</head>

includes/footer.php
	
	</main>
	
	<script>
	
	
	  function increaseValue() {
	    var value = parseInt(document.getElementById('number').value, 10);
	    value = isNaN(value) ? 0 : value;
	    value++;
	    document.getElementById('number').value = value;
	    $('#decrease').addClass('activo');
	  }

	  function decreaseValue() {
	    var value = parseInt(document.getElementById('number').value, 10);
	    value = isNaN(value) ? 0 : value;
	    value < 1 ? value = 1 : '';
	    value--;
	    document.getElementById('number').value = value;
	  } 
	  function increaseValue1() {
	    var value = parseInt(document.getElementById('number1').value, 10);
	    value = isNaN(value) ? 0 : value;
	    value++;
	    document.getElementById('number1').value = value;
	    $('#decrease1').addClass('activo');
	  }

	  function decreaseValue1() {
	    var value = parseInt(document.getElementById('number1').value, 10);
	    value = isNaN(value) ? 0 : value;
	    value < 1 ? value = 1 : '';
	    value--;
	    document.getElementById('number1').value = value;
	  } 
	  function increaseValue2() {
	    var value = parseInt(document.getElementById('number2').value, 10);
	    value = isNaN(value) ? 0 : value;
	    value++;
	    document.getElementById('number2').value = value;
	    $('#decrease2').addClass('activo');
	  }

	  function decreaseValue2() {
	    var value = parseInt(document.getElementById('number2').value, 10);
	    value = isNaN(value) ? 0 : value;
	    value < 1 ? value = 1 : '';
	    value--;
	    document.getElementById('number2').value = value;
	  } 
	  function increaseValue3() {
	    var value = parseInt(document.getElementById('number3').value, 10);
	    value = isNaN(value) ? 0 : value;
	    value++;
	    document.getElementById('number3').value = value;
	    $('#decrease3').addClass('activo');
	  }

	  function decreaseValue3() {
	    var value = parseInt(document.getElementById('number3').value, 10);
	    value = isNaN(value) ? 0 : value;
	    value < 1 ? value = 1 : '';
	    value--;
	    document.getElementById('number3').value = value;
	  } 
	</script>

	<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.2.4/jquery.min.js" integrity="sha512-DUC8yqWf7ez3JD1jszxCWSVB0DMP78eOyBpMa5aJki1bIRARykviOuImIczkxlj1KhVSyS16w2FSQetkD4UU2w==" crossorigin="anonymous"></script>
	<script>window.jQuery || document.write('<script src="js/jquery-2.2.4.min.js"><\/script>');</script>

	<?php
		Plugins::cargar();
		$main_js = $minified->merge(BASE_PATH.'js/main.min.js', 'js', array(BASE_PATH.'js/main.js'));
		echo '<script src="'.str_replace(BASE_PATH, '', $main_js).'?v='.filemtime($main_js) .'"></script>'. PHP_EOL;
   ?>
</body>
</html>

includes/header.php
<?php include('head.php'); ?>

<body class="<?= 's_'.SECCION ?>">

<header class="cabecera" id="home">
   <div class="container">
     <div class="row">
       <div class="buscador">
        <a href=""><span class="material-icons">search</span></a>
         <input type="text" placeholder="Qué esta pasando?" value="">
       </div>
       <div class="logo">
        <a href="">  <?php include('images/wabi.svg'); ?> </a>
       </div>
       <div class="menu">
          <ul class="lista">
            <li class="lenguaje">
              <div class="js-desplegable idioma">
                <a class="js-desplegable-enlace">
                  <?php switch (IDIOMA) {
                  case 'es':
                  echo "ES";
                  break;
                  case 'en':
                  echo "EN";
                  break;

                  } ?>
                <span class="material-icons don">expand_more</span>
                </a>
                <ul class="js-desplegable-contenido" >
                  <li><a href="<?=BASE_URL?>en/">EN</a></li>
                  <li><a href="<?=BASE_URL?>es/">ES</a></li>
                  <li><a href="<?=BASE_URL?>es/">GR</a></li>
                  <li><a href="<?=BASE_URL?>es/">MAL</a></li>
                  <li><a href="<?=BASE_URL?>es/">ITA</a></li>
                  <li><a href="<?=BASE_URL?>es/">RU</a></li>
                  <li><a href="<?=BASE_URL?>es/">PR</a></li>
                  <li><a href="<?=BASE_URL?>es/">ARA</a></li>
                  <li><a href="<?=BASE_URL?>es/">HIN</a></li>
                </ul>
              </div>
            </li>
          </ul> 
          <a href=""><span class="material-icons-outlined">person_outline</span></a>
          <a href="" class="cart">
            <span class="material-icons-outlined">shopping_cart</span>
            <p>0</p>
          </a>
       </div>
     </div>
   </div>
   <div class="banda-verde">
     <div class="container">
       <div class="row">
         <ul>
           <li>
            <a href="" class="<?= SECCION == 'home' ? 'activo' : '' ?>"> promociones</a> 
           </li>
            <li>
             <a href="">alcohol</a>
           </li>
            <li>
             <a href="">beverages</a>
           </li>
            <li>
             <a href="">groceries</a>
           </li>
            <li>
             <a href="">beauty and healthcare</a>
           </li>
           <li>
            <a href="">home cleaning</a>
           </li>
         </ul>
       </div>
     </div>
   </div>
</header>
<div class="phone">
  <input type="radio" name="s" id="s1"  >
  <input type="radio" name="s" id="s2">
  <input type="radio" name="s" id="s3">
  <input type="radio" name="s" id="s4" checked="checked">
  <label for="s1"><span class="material-icons icos">search</span></label>
  <label for="s2"><span class="material-icons-outlined icos">person_outline</span></label>
  <label for="s3"> <span class="material-icons-outlined icos">shopping_cart</span></label>
  <label for="s4">
    <button class="hamburger hamburger--efecto visible-xs">
        <span class="hamburger-box">
          <span class="hamburger-inner"></span>
        </span>
      </button>

  </label>
  <div class="circle"></div>
  <div class="phone_content">
    <div class="phone_bottom">
      <span class="indicator"></span>
    </div>
  </div>
</div>
<main>

js/main.js
$(document).ready(function(){

	/** 
	* PLUGINS
	* Todo el JS está dividido en módulos que cargamos opcionalmente junto con
	* sus dependencias (otros JS y CSS).
	*
	* Con la función cargarPlugin() definimos si se carga o no y con cargarRecursos()
	* cargamos dinámicamente todas las dependencias del script que queremos ejecutar.
	*
	* CargarPlugin() lee un array creado con el método Plugins::activar() del helper 
	* Plugins (core/Plugins.php)
	*
	* Ej :  Plugins::activar('owlCarousel')
	*/
	
	// Tabs
		// https://github.com/filamentgroup/Accessible-jQuery-Tabs
	if( cargarPlugin('tabs') ){
		cargarRecursos([
			{tipo: 'js', src: 'js/jquery.tabs/jquery.tabs.min.js'},
		], function(){
			$(window).resize(function(){
				if( $(window).width() > 990){
					if(window.tabs === undefined){
						window.tabs = $('.tab-seccion').tabs({
							autoRotate: 2500,
							alwaysScrollToTop:false
						});
					}
				}else{
					if(window.tabs !== undefined){
						$('.tab-seccion').destroyTabs();
						window.tabs = undefined;
					}
				}
			}).resize();
		});
		if( $(window).width() < 990){
			// $(".js-cont-desplegable").hide();
			$(".mobile-tittle").click(function(e){
				$(".mobile-tittle").removeClass('selected');
				$(this).addClass('selected');
		       var desplegable = $(this).next();
		       $('.js-cont-desplegable').not(desplegable).slideUp('fast');
		        desplegable.slideToggle('fast');		    	
	    	});
	    }
	}
	// Galerías (Owl Carousel)
	// http://owlcarousel2.github.io/OwlCarousel2/
	if( cargarPlugin('owlCarousel') ){
		cargarRecursos([
			{tipo: 'css', src: 'js/jquery.owl-carousel/assets/owl.carousel.min.css'},
			{tipo: 'css', src: 'js/jquery.owl-carousel/assets/animate.min.css'},
			{tipo: 'js',  src: 'js/jquery.owl-carousel/owl.carousel.min.js'},

		], function(){

			$('.slider-principal > ul').owlCarousel({
				items: 3,
			    loop: true,
				autoplay: true,
			    autoplayTimeout: 6000,
			    slideSpeed: 100,
		        nav: true,
		        dots: true,
		        dotsContainer: '.dotsCont',
		        navText: [
	            "<i class=\"icon-left-open icon\"></i><span class=\"nav-slider-prev\">Anterior</span>",
	            "<i class=\"icon-right-open icon\"></i><span class=\"nav-slider-next\">Siguiente</span>"
	            ],
	             responsive:{
                0:{
                    items:1
                },
                550:{
                    items:2
                },
                767:{
                    items:3
                },
              },
			});	

			$('.slider-novedad > ul').owlCarousel({
				autoplay: false,
				items: 1,
			    loop: true,
			    slideSpeed: 1000,
			    autoplaySpeed: 500,
		        singleItem: true,
		        dragBeforeAnimFinish: false,
		        mouseDrag: false,
		        touchDrag: true,
		        nav: true,
		        dots: false,
		        navText: [
	            "<i class=\"icon-left-open icon\"></i><span>Anterior</span>",
	            "<i class=\"icon-right-open icon\"></i><span>Siguiente</span>"
	            ],
			});		

		});
	}



	// Gridder
	// https://github.com/oriongunning/gridder
	if( cargarPlugin('gridder') ){
		cargarRecursos([
			{tipo: 'js',  src: 'js/jquery.gridder/jquery.gridder.min.js'},

		], function(){
				$('.gridder').gridderExpander({
				    scroll: false,
				    scrollOffset: 30,
				    scrollTo: "panel",                  // panel or listitem
				    animationSpeed: 400,
				    animationEasing: "easeInOutExpo",
				    showNav: false,                      // Show Navigation
				    nextText: "Next",                   // Next button text
				    prevText: "Previous",               // Previous button text
				    closeText: "Close",                 // Close button text
				    onStart: function(){
				        //Gridder Inititialized
				    },
				    onContent: function(){
				        //Gridder Content Loaded
				    },
				    onClosed: function(){
				        //Gridder Closed
				    }
				});

				// Abrir el primero por default
				if( cargarPlugin('abrirGridder') ){
					$('[data-abierto=1]').trigger('click');
				}
		});
	}
	// }


	// WOW (animación)
	// https://github.com/matthieua/WOW
	if( cargarPlugin('animacion') ){
		cargarRecursos([
			{tipo: 'css',  src: 'css/animate.min.css'},
			{tipo: 'js',  src: 'js/jquery.wow/wow.min.js'},

		], function(){
			wow = new WOW({
				boxClass:     'wow',     
				animateClass: 'animated',
				offset:       0,         
				mobile:       true,      
				live:         false       
		    }).init();
		});
	}

	



	/** 
	* JS GENERALES
	* Estos scripts se ejecutan siempre.
	*
	*/

	// Transiciones entre páginas y animación de enlaces internos # ///

	




	 
	/// Scroll en cabecera ///
	// Anima el menú cuando hay scroll
	$( window ).scroll(function(){
		var $cabecera = $('.cabecera');
		if( $(window).scrollTop() > 20){
			$cabecera.addClass('scroll');
		}else{
			$cabecera.removeClass('scroll');
		}
	});
	// Oculta y muestra el menú cuando hay scroll
	var $cabecera = $('.cabecera');
	var previousScroll = 0;
	$(window).scroll(function(event){
	   var scroll = $(this).scrollTop();
	   if (scroll > previousScroll && scroll > 400){
	       $cabecera.addClass('ocultar');
	       //Cierra el menú cuando hay scroll
			$(".navbar-collapse").removeClass("in").addClass("collapse");
			$(".hamburger").removeClass("is-active");
	   } else {
	      $cabecera.removeClass('ocultar');
	   }
	   previousScroll = scroll;
	});

	/// Menú hamburguesa ///


	$(document).ready(function(){
		var $side = $('.banda-verde');
		var $hamburguesa = $('.hamburger');
		var $phone = $('.circle');
		$('.hamburger').click(function(){
			$hamburguesa.toggleClass('is-active');
			$side.toggleClass('activo');

		});
		$('.lista').click(function() {
	    $('.js-desplegable-contenido').toggleClass('block');
	    $('.don').toggleClass('girar');
		});
	});

	$( window ).scroll(function(){
			var $cabecera = $('.cabecera');
			var $side = $('.banda-verde');
			var $hamburguesa = $('.hamburger');

			
			if( $(window).scrollTop() > 1){
				$side.removeClass('activo');
				$hamburguesa.removeClass('is-active');
			}else{
				$side.addClass('banda-verde');
			}
		});


	 
	/** 
	* HELPERS
	* Estas funciones son las que usamos definir si cargar o no el resto del JavaScript
	* y cargar sus recursos
	*
	*/

	// Reviso si cargar un plugin JS
	// Cargo solo los plugins listados en el array jsPlugins (El array jsPlugins se genera con PHP).
	function cargarPlugin(plugin){
		return (window.jsplugins.indexOf( plugin ) !== -1);
	}

	// Cargo los recursos dinámicamente y ejecuto el callback
	// (Pueden ser CSS o JS)
	function cargarRecursos(recursos, callback) {
		var total = recursos.length;
		recursos.forEach(function(r){
			var s;
			if(r.tipo == 'css'){
				s = document.createElement( 'link' );
				s.setAttribute('rel','stylesheet');
				s.setAttribute('type','text/css');
				s.setAttribute('href',r.src);
			}
			if(r.tipo == 'js'){
				s = document.createElement( 'script' );
				s.setAttribute( 'src', r.src );
			}
			s.onload = s.onreadystatechange = function() {
			    if(!this.readyState || this.readyState == "loaded" || this.readyState == "complete") {
			        total--;
					/*console.log('Cargó ' + r.src);*/
    				if(total === 0){
    					callback();
    				}
			    }
			};
			document.head.insertBefore(s, document.head.firstChild);
		});
	}
});


	// Modal Version Beta

// document.addEventListener('DOMContentLoaded', modal);

// function modal () {

//   document.getElementById("cerrarBtn").addEventListener("click", cerrar);
//   document.body.addEventListener("click", cerrar);
//   window.addEventListener("keydown", cerrar);

//   function cerrar () {
//     document.getElementById("popup").style.display = "none";
//     document.getElementById("sombra").style.display = "none";
//     document.body.style.overflowY = "scroll";
//   }
// }